external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/CachedKafkaConsumer.scala

```diff
@@ -83,6 +86,113 @@ private[kafka010] case class CachedKafkaConsumer private(
     record
   }
 
+  @tailrec
+  final def getAndIgnoreLostData(
+      offset: Long,
+      untilOffset: Long,
+      pollTimeoutMs: Long): ConsumerRecord[Array[Byte], Array[Byte]] = {
+    // scalastyle:off
+    // When `failOnDataLoss` is `false`, we need to handle the following cases (note: untilOffset and latestOffset are exclusive):
+    // 1. Some data are aged out, and `offset < beginningOffset <= untilOffset - 1 <= latestOffset - 1`
+    //      Seek to the beginningOffset and fetch the data.
+    // 2. Some data are aged out, and `offset <= untilOffset - 1 < beginningOffset`.
+    //      There is nothing to fetch, return null.
+    // 3. The topic is deleted.
+    //      There is nothing to fetch, return null.
+    // 4. The topic is deleted and recreated, and `beginningOffset <= offset <= untilOffset - 1 <= latestOffset - 1`.
+    //      We cannot detect this case. We can still fetch data like nothing happens.
+    // 5. The topic is deleted and recreated, and `beginningOffset <= offset < latestOffset - 1 < untilOffset - 1`.
+    //      Same as 4.
+    // 6. The topic is deleted and recreated, and `beginningOffset <= latestOffset - 1 < offset <= untilOffset - 1`.
+    //      There is nothing to fetch, return null.
+    // 7. The topic is deleted and recreated, and `offset < beginningOffset <= untilOffset - 1`.
+    //      Same as 1.
+    // 8. The topic is deleted and recreated, and `offset <= untilOffset - 1 < beginningOffset`.
+    //      There is nothing to fetch, return null.
+    // scalastyle:on
+    if (offset >= untilOffset) {
+      // Case 2 or 8
+      // We seek to beginningOffset but beginningOffset >= untilOffset
+      reset()
+      return null
+    }
+
+    logDebug(s"Get $groupId $topicPartition nextOffset $nextOffsetInFetchedData requested $offset")
+    var outOfOffset = false
```

Comment:
Can this var be eliminated by just using a single try around the if / else?  It's the same catch condition in either case